contrib/aerospike/aerospike-client.v5/aerospike.go

```diff
@@ -0,0 +1,420 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package aerospike provides functions to trace the aerospike/aerospike-client-go package (https://github.com/aerospike/aerospike-client-go).
+//
+// `WrapClient` will wrap an aerospike `Client` and return a new struct with all
+// the same methods, so should be seamless for existing applications. It also
+// has an additional `WithContext` method which can be used to connect a span
+// to an existing trace.
+package aerospike // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/aerospike/aerospike-client.v5"
+
+import (
+	"context"
+	"math"
+	"time"
+
+	as "github.com/aerospike/aerospike-client-go/v5"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+// WrapClient wraps an aerospike.Client so that all requests are traced using the
+// default tracer with the service name "aerospike".
+func WrapClient(client *as.Client, opts ...ClientOption) *Client {
+	cfg := new(clientConfig)
+	defaults(cfg)
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	log.Debug("contrib/aerospike/aerospike-client.v5: Wrapping Client: %#v", cfg)
+	return &Client{
+		Client:  client,
+		cfg:     cfg,
+		context: context.Background(),
+	}
+}
+
+// A Client is used to trace requests to the aerospike server.
+type Client struct {
+	*as.Client
+	cfg     *clientConfig
+	context context.Context
+}
+
+// WithContext creates a copy of the Client with the given context.
+func (c *Client) WithContext(ctx context.Context) *Client {
+	// the existing aerospike client doesn't support context, but may in the
+	// future, so we do a runtime check to detect this
+	mc := c.Client
+	if wc, ok := (interface{})(c.Client).(interface {
+		WithContext(context.Context) *as.Client
+	}); ok {
+		mc = wc.WithContext(ctx)
+	}
```

Comment:
~TIL! I don't remember seeing this trick before!~ Edit: But we have used it [before](https://github.com/DataDog/dd-trace-go/blob/d86ccba9b41cc46a3a6b0138eaf7f99a829a3edd/contrib/bradfitz/gomemcache/memcache/memcache.go#L54). Nice! 🙂 

Why are you checking if `WithContext` exists? Is it because it's not there in all versions? I'd assume yes, but just making sure. Maybe this needs to be answered for everyone

```suggestion
	if wc, ok := (interface{})(c.Client).(interface {
		WithContext(context.Context) *as.Client
	}); ok {
		// we have support for WithContext; propagate the call
		mc = wc.WithContext(ctx)
	}
```

Reply:
Same as for Memcache, it's for future proofing. The aerospike client doesn't support any context stuff, but if it does at some point the instrumentation will opt-in. That's assuming it's done with a `WithContext()` function.
This is not my design though, I merely follow the Memcache design from this repo.

Reply:
Perfectly fine. Just adding the comment will be good enough.